engine/events/odd_egg.asm

```diff
@@ -39,11 +39,13 @@ _GiveOddEgg:
 .done
 
 	ld hl, OddEggs
-	ld a, OddEgg1End - OddEgg1
+	ld a, NICKNAMED_MON_STRUCT_LENGTH
 	call AddNTimes
 
-	ld de, wOddEggSpecies
-	ld bc, PARTYMON_STRUCT_LENGTH + 2 * MON_NAME_LENGTH
+	; Writes to wOddEgg, wOddEggName, and wOddEggOTName,
+	; even though OddEggs does not have data for wOddEggOTName
+	ld de, wOddEgg
+	ld bc, NICKNAMED_MON_STRUCT_LENGTH + MON_NAME_LENGTH
```

Comment:
This should be `NAME_LENGTH`. Please change that in `wram.asm`'s `wOddEggOTName` as well.